### PR TITLE
Issue 89 - Integrate IncompleteCheckout model 

### PR DESCRIPTION
**Pull request by ahmed-shehata**
Issue #89 Integrate IncompleteCheckout model

### CODE REVIEW

**Comment by ahmed-shehata** [comment]
👍

**Comment by ahmed-shehata** [comment]
👍

**Comment by ahmed-shehata** [comment]
👍

File: AtlasSDK/AtlasSDK/Models/Checkout.swift [hunk 1]
```diff
@@ -4,19 +4,17 @@
...
-    public let billingAddressId: String?
-    public let shippingAddressId: String?
-    public let billingAddress: BillingAddress?
-    public let shippingAddress: ShippingAddress?
+
     public let delivery: Delivery
     public let payment: Payment
 
-    public var availableAddresses: [Address]?
+    public let billingAddress: BillingAddress?
```
Comment (haldun) [inline AtlasSDK/AtlasSDK/Models/Checkout.swift]: If this struct represents a valid Checkout, then these fields must be non-optional.

File: AtlasSDK/AtlasSDK/PublicAPIClient.swift [hunk 2]
```diff
@@ -25,7 +25,7 @@ public typealias CartCompletion = AtlasResult<Cart> -> Void
 /**
  Completion block `AtlasResult` with the `Checkout` struct as a success value
  */
-public typealias CheckoutCompletion = AtlasResult<Checkout> -> Void
+public typealias CheckoutCompletion = AtlasResult<CheckoutType> -> Void
```
Comment (haldun) [inline AtlasSDK/AtlasSDK/PublicAPIClient.swift]: Since API returns a real `Checkout`, this should be not a `CheckoutType` but `Checkout` itself.

**Comment by bonkey** [comment]
@haldun we'll read your remarks and we'll consider them, but probably tomorrow. for now i'd merge and refactor in second round tomorrow

**Comment by bonkey** [comment]
:+1:

**Comment by bonkey** [comment]
haha... 
> Pull requests that have a failing status can’t be merged from a phone.

**Pull request merged**